### PR TITLE
change: remove listen.host from api/conf/conf.yaml

### LINKED ISSUE #1743: change: remove `listen.host` from api/conf/conf.yaml
# Feature request

## Please describe your feature

The configuration item `listen.host` in conf.yaml often confuses users, because when the configuration dashboard can be accessed non-locally, two places need to be modified (the other place is `allow_list`).
We can remove the `listen.host` configuration item from conf.yaml, but retain the feature, and users can add it when they need it.

## Describe the solution you'd like

1. remove `listen.host` from api/conf/conf.yaml
2. set `ServerHost` in [the code](https://github.com/apache/apisix-dashboard/blob/master/api/internal/conf/conf.go#L47
) to an empty string

### PR DESCRIPTION
**Pull request by nic-chen**
Please answer these questions before submitting a pull request, **or your PR will get closed**.

**Why submit this pull request?**

- [ ] Bugfix
- [ ] New feature provided
- [ ] Improve performance
- [ ] Backport patches

The configuration item `listen.host` in conf.yaml often confuses users, because when the configuration dashboard can be accessed non-locally, two places need to be modified (the other place is allow_list).
remove the `listen.host` configuration item from conf.yaml, but retain the feature, and users can add it when they need it.

**What changes will this PR take into?**

remove listen.host from api/conf/conf.yaml
set ServerHost in the code to an empty string

**Related issues**

fix/resolve #1743

**Checklist:**

- [x] Did you explain what problem does this PR solve? Or what new features have been added?
- [x] Have you added corresponding test cases?
- [ ] Have you modified the corresponding document?
- [x] Is this PR backward compatible? If it is not backward compatible, please discuss on the mailing list first

### CODE REVIEW

**Comment by juzhiyuan** [comment]
bugfix or feature?

**Comment by juzhiyuan** [comment]
may sync master codes to fix CI

**Comment by juzhiyuan** [comment]
Please sync the master branch

**Review by starsz — COMMENTED** [review]
Why not set “0.0.0.0” as default？

File: .github/workflows/deploy-with-docker.yml [hunk 1]
```diff
@@ -33,8 +33,9 @@ jobs:
       - name: Modify Config
         run: |
           sed -i 's/127.0.0.1:2379/172.16.238.10:2379/' api/conf/conf.yaml
-          sed -i 's/host: 127.0.0.1/host: 0.0.0.0/' api/conf/conf.yaml
```
Comment (tokers) [inline .github/workflows/deploy-with-docker.yml:36]: Should add some notes about the default listen IP address(es).

**Comment by nic-chen** [comment]
> Why not set “0.0.0.0” as default？

Because in our configuration, `listen` will only have  `port`.

So it would be strange to display `0.0.0.0:9000` after the `manager-api` started, `:9000` should be better. 

What do you think?  @starsz

File: .github/workflows/deploy-with-docker.yml (hunk 1 above)
Comment (nic-chen) [reply .github/workflows/deploy-with-docker.yml:36]: added.

**Review by tokers — APPROVED** [review]

File: api/conf/conf.yaml [hunk 2]
```diff
@@ -17,10 +17,15 @@
 
 conf:
   listen:
-    host: 127.0.0.1     # `manager api` listening ip or host name
```
Comment (starsz) [inline api/conf/conf.yaml:20]: We still support configuring the host. So can we comment on this instead of removing them?

Comment (membphis) [reply api/conf/conf.yaml:20]: @starsz your way is better

File: api/conf/conf.yaml [hunk 3]
```diff
@@ -17,10 +17,15 @@
...
     port: 9000          # `manager api` listening port
   allow_list:           # If we don't set any IP list, then any IP access is allowed by default.
     - 127.0.0.0/24
+    - 10.0.0.0/8
```
Comment (membphis) [inline api/conf/conf.yaml:23]: only one `127.0.0.0/24` is enough.

we can add more comment to show how to set it

**Review by gxthrj — APPROVED** [review]

File: api/conf/conf.yaml (hunk 3 above)
Comment (nic-chen) [reply api/conf/conf.yaml]: updated.

File: api/conf/conf.yaml (hunk 2 above)
Comment (nic-chen) [reply api/conf/conf.yaml:20]: updated.

**Review by starsz — APPROVED** [review]

**Comment by starsz** [comment]
Emmm.Need to fix the link error. @nic-chen

**Comment by nic-chen** [comment]
> Emmm.Need to fix the link error. @nic-chen

sure

**Pull request merged**